tests/e2e/hcp_cluster_test.go

```diff
@@ -0,0 +1,177 @@
+				Expect(role).To(Equal(role))
+
+			})
+
+		It("create cluster with the KMS and etcd encryption  for hypershift clusters by rosa-cli - [id:60083]",
+			labels.High, labels.NonClassicCluster,
+			func() {
+				By("Check the help message of 'rosa create cluster -h'")
+				output, err := clusterService.RetrieveHelpForClusterCreate()
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring("--kms-key-arn"))
+				Expect(output.String()).To(ContainSubstring("--etcd-encryption"))
+				Expect(output.String()).To(ContainSubstring("--enable-customer-managed-key"))
+
+				By("Get cluster description")
+				jsonData, err := clusterService.GetJSONClusterDescription(clusterID)
+				Expect(err).To(BeNil())
+
+				if !isHosted {
+					Skip("It does't non HCP cluster - cannot run this test")
+				}
+
+				ectdKMS := jsonData.DigString("aws", "etcd_encryption", "kms_key_arn")
+				npKMS := jsonData.DigString("aws", "kms_key_arn")
+
+				if jsonData.DigBool("aws", "etcd_encryption") {
+					Expect(ectdKMS).ToNot(BeEmpty())
+				}
+				Expect(jsonData.DigString("aws", "kms_key_arn")).To(Equal(npKMS))
+			})
+
+		It("create HCP cluster with network type can work well via rosa cli - [id:71050]",
+			labels.High, labels.NonClassicCluster,
+			func() {
+				By("Check the help message of 'rosa create cluster -h'")
+				//It is hiddened now
+				helpOutput, err := clusterService.RetrieveHelpForClusterCreate()
+				Expect(err).To(BeNil())
+				Expect(helpOutput.String()).ToNot(ContainSubstring("--no-cni"))
+
+				By("Get cluster description")
+				if !isHosted {
+					Skip("It does't non HCP cluster - cannot run this test")
+				}
+				output, err := clusterService.DescribeCluster(clusterID)
+				Expect(err).To(BeNil())
+				clusterDetail, err := clusterService.ReflectClusterDescription(output)
+				Expect(err).To(BeNil())
+				networkLine := clusterDetail.Network[0]
+
+				By("Get cluster description via json")
+				jsonData, err := clusterService.GetJSONClusterDescription(clusterID)
+				Expect(err).To(BeNil())
+				Expect(networkLine["Type"]).To(Equal(jsonData.DigString("network", "type")))
+			})
+
+		It("create ROSA HCP cluster with external_auth_config config should work well via rosa client - [id:71945]",
+			labels.High, labels.NonClassicCluster,
+			func() {
+				By("Check the help message of 'rosa create cluster -h'")
+				helpOutput, err := clusterService.RetrieveHelpForClusterCreate()
+				Expect(err).To(BeNil())
+				Expect(helpOutput.String()).To(ContainSubstring("--external-auth-providers-enabled"))
+
+				By("Get cluster description")
+				jsonData, err := clusterService.GetJSONClusterDescription(clusterID)
+				Expect(err).To(BeNil())
+
+				if !isHosted {
+					Skip("It does't non HCP cluster - cannot run this test")
+				}
+				output, err := clusterService.DescribeCluster(clusterID)
+				Expect(err).To(BeNil())
+				clusterDetail, err := clusterService.ReflectClusterDescription(output)
```

Comment:
Please make external auth supported first.

Reply:
Added it in the pr